src/java.base/share/classes/java/lang/module/ModuleReader.java

```diff
@@ -213,9 +213,17 @@ default void release(ByteBuffer bb) {
      * when using the stream to list the module contents and access is denied
      * by the security manager. </p>
      *
+     * <p> The returned stream contains references to one or more open directories
+     * in the module. The directories are closed by closing the stream. </p>
+     *
      * <p> The behavior of the stream when used after the module reader is
      * closed is implementation specific and therefore not specified. </p>
      *
+     * @apiNote
+     * This method must be used within a try-with-resources statement or similar
+     * control structure to ensure that the stream's open directories are closed
```

Comment:
Similar to my other comment - should "must' be replaced with "should" - for the same aforementioned reason.